### ISSUE TITLE
when add BareMetal host, new ui misses 4 parameters: cpu cores, cpu MHz,memory,host mac

### ISSUE THREAD

**Issue by hooklee2000** [apache/cloudstack#5772]
##### ISSUE TYPE
  Bug Report
##### COMPONENT NAME
new UI
~~~

~~~

4.15.1, Maybe 4.16.0 also (check  cloudstack/ui/src/views/infra/HostAdd.vue )

~~~

~~~

##### OS / ENVIRONMENT
CentOS 7.4

##### SUMMARY
Using the new ui,we can'nt add baremetal host without 4 parameters: cpu cores, cpu MHz,memory,mac.
But the legacy ui can add baremetal host with 4 parameters.

![图片](https://user-images.githubusercontent.com/52850777/145929449-aac25b35-fd5d-48b7-84e3-ad1226f9fcd2.png)

**Comment by utchoang** [comment]
@shwstppr @Pearl1594 As I see in the API description https://cloudstack.apache.org/api/apidocs-4.16/apis/addHost.html, where should these parameters be set?

**Comment by shwstppr** [comment]
@utchoang not sure tbh. Checked legacy UI and it passes them as the `addHost` API params. Although the API doesn't have any such params.
```
                                    if (selectedClusterObj.hypervisortype == "BareMetal") {
                                        $.extend(data, {
                                            cpunumber: args.data.baremetalCpuCores,
                                            cpuspeed: args.data.baremetalCpu,
                                            memory: args.data.baremetalMemory,
                                            hostmac: args.data.baremetalMAC
                                        });
                                    } else if (selectedClusterObj.hypervisortype == "Ovm") {
                                        $.extend(data, {
                                            agentusername: args.data.agentUsername,
                                            agentpassword: args.data.agentPassword
                                        });
                                    } else if (selectedClusterObj.hypervisortype == "Ovm3") {
                                        $.extend(data, {
                                            agentusername: args.data.agentUsername,
                                            agentpassword: args.data.agentPassword,
                                            agentport: args.data.agentPort
                                       });
                                    }
                                }

                                var hostId = null;
                                $.ajax({
                                    url: createURL("addHost"),
                                    type: "POST",
                                    data: data,
                                    success: function (json) {
```

there is a `addBaremetalHost` API in code but even that doesn't have these params

**Comment by harikrishna-patnala** [comment]
as far as I remember these values (CPU, memory, cores and mac) are passed in details Parameter. Since there is no hypervisor layer, we have to feed these values while adding the baremetal host.

**Comment by shwstppr** [comment]
@harikrishna-patnala don't see any `details` param in the API in 4.16

**Comment by harikrishna-patnala** [comment]
Yes I just checked, neither detaills parameter or specific name parameters are not defined in API. I've further checked it looks like it using the "fullUrlParams" from the BaseCmd like this https://github.com/apache/cloudstack/blob/2e9c9417be150384db6f503dff90dbbfeeeae09e/server/src/main/java/com/cloud/resource/ResourceManagerImpl.java#L641

and then use these params to fetch the required details like showing here,

https://github.com/apache/cloudstack/blob/6e47c49fbf5ae4a7ab521409dd6fd4751189f902/plugins/hypervisors/baremetal/src/main/java/com/cloud/baremetal/manager/BareMetalDiscoverer.java#L186-L189

**Comment by utchoang** [comment]
@harikrishna-patnala @shwstppr So I will still pass these 4 parameters to the addHost api and it will work right?

**Comment by harikrishna-patnala** [comment]
Yes @utchoang, I have not tested recently but that is how I see the code too. I found the below statement from one of the documents, which says these params are required
_"add a host, provide all information required by UI page. Note the 'hostname' is ipmi ip, 'account' is ipmi account, 'password' is ipmi password. Please make sure cpu number/speed, memory size is matching your real host fact, otherwise the dashboard may not display precisely. IMPORTANT: the host tag must be set to some value, otherwise the host will not be found by host allocator. If multiple host tags are set, only the first one takes effect_
create a service offering with the same host tag specified in step 7
"

**Comment by sureshanaparti** [comment]
Fixed in #5812
